### ISSUE TITLE
Auto Morph to Event Sub-Process by changing Start Event Type

### ISSUE THREAD

**Issue by falko** [camunda/camunda-modeler#915]
Given a Sub-Process with a None Start Event. When I morph the Start Event into a different Start Event type, e.g. a Message Start Event, I would like the Sub-Process to be auto-morphed into an Event Sub-Process.

I just finished a modeling session in front of 10 people of a large customer. The resulting model had half a dozen Event Sub-Processes. In such an expensive meeting every click saved results in more valuable discussion time.

Many customers are also asking how they can create an Event Sub-Process because it is not intuitive.

Furthermore, a 'normal' Sub-Process with a Start Event other than a None Start Event is a BPMN Syntax Error. Usually, the modeler is quite good at preventing them, but here is a loophole to create an invalid model.

Consequently, we would auto-morph back into a normal Sub-Process when the Start Event is morphed into a None Start Event.

Auto-morphing the Start Event, when the Sub-Process is morphed, would also be nice. However, morphing to an Event Sub-Process would require a sensible default auto-morph type for the Start Event, e.g. Message Start Event.

Chaining this with #914 would be perfect, i.e. Drawing a Message Flow to a Sub-Process with None Start Event makes it an Event Sub-Process with a Message Start Event.

Would this be an easy-pick for a pull request?

**Comment by philippfromme** [comment]
This would be implemented as a behavior that implicitly replaces the sub-process after replacing the event. Not an easy pick but doable.

**Comment by falko** [comment]
I still wish we had this literally every time I create an event sub-process. The only thing that eases the pain is that all the extra clicks have kind-of gone into muscle memory by now ;-)

**Comment by falko** [comment]
But beginners can obviously not rely on muscle memory and may have a hard time even finding out how to create their first event sub-process if they are like me focusing on the event-type first and not caring about BPMN requiring a dotted line around the sub-process.

**Comment by nikku** [comment]
Thanks for bringing this up again. It would solve some issues we have currently, including the fact that certain things you can model today are actually illegal:

![image](https://user-images.githubusercontent.com/58601/216315326-4cb15e96-570d-49f4-9056-5650cee1342f.png)

We want to prevent illegal intermediate steps, and [assist the users at the same time](https://github.com/bpmn-io/design-principles#do-the-hard-work-to-make-it-simple)

**Comment by nikku** [comment]
What is clearly visible in our existing behavior:

* We explicitly limit what the users can do (rather than auto-morphing)
* Changing to an event-sub-process is a 2+ step operation (rather than one)

----

![capture OvU7oX_optimized](https://user-images.githubusercontent.com/58601/216316216-16f04690-065a-4e96-a891-f1de3f9e828c.gif)

**Comment by barmac** [comment]
I don't think this is an easy pick. What should happen if my embedded subprocess has a boundary event? Should we discard the boundary event? I think that would be surprising to the user. I'd rather keep the current behavior where you switch to event subprocess explicitly.

**Comment by barmac** [comment]
Note that with create-append-anything you can already append an event subprocess:

https://github.com/user-attachments/assets/106866b8-4e8d-43d7-ae17-13a33aa06e69

**Comment by falko** [comment]
How about we only do it where the desire is clear, i.e. we offer the other start event types only when the preconditions for an event sub-process are met, e.g. it has no boundary events and neither incoming nor outgoing sequence flows? That would be a UX improvement for all users creating event sub-processes while still avoiding accidents for other cases.

**Comment by nikku** [comment]
> How about we only do it where the desire is clear, i.e. we offer the other start event types only when the preconditions for an event sub-process are met, e.g. it has no boundary events and neither incoming nor outgoing sequence flows? That would be a UX improvement for all users creating event sub-processes while still avoiding accidents for other cases.

We have a set of [design principles](https://github.com/bpmn-io/design-principles) that govern how we build our libraries. One of the principle is _less is more_, this prevents us from creating too many special cases, each of these must be learned.

The question to answer if this special case is worth the effort, especially as nowadays with create/append anything (cf. https://github.com/camunda/camunda-modeler/issues/915#issuecomment-2706777732) an event sub-process can trivially be created:

![Image](https://github.com/user-attachments/assets/2e5cec02-34b1-4ee1-917a-fd137607068c)

**Comment by philippfromme** [comment]
> we offer the other start event types only when the preconditions for an event sub-process are met

This is against [_No surprises_](https://github.com/bpmn-io/design-principles?tab=readme-ov-file#no-surprises). I can already see the bug reports coming in. 😅
